File: 백준/Silver/24479. 알고리즘 수업 － 깊이 우선 탐색 1/알고리즘 수업 － 깊이 우선 탐색 1.py
import sys
input = sys.stdin.readline
sys.setrecursionlimit(10**9)


buffer = input().split()
n = int(buffer[0])
m = int(buffer[1])
r = int(buffer[2])

graph= [ [] for _ in range(n+1)]


for i in range(m):
    buffer = input().split()
    a = int(buffer[0])
    b = int(buffer[1])

    graph[a].append(b)
    graph[b].append(a)


for i in range(n+1):
    graph[i].sort()


visited = [ 0 for _ in range(n+1) ]


# def dfs(v, r, c) : 
#    visited[r] = c + 1
#    print("방문",r)
#    for i in graph[r]:
#        if visited[i] == 0 :
#            dfs(v, i, visited[r])


# dfs(n, r, 0)

# for i in range(1, n+1):
#     print(visited[i])
    

cnt=[0]*(n+1)

def dfs(v,e,r):
    global count 
    visited[r]=1
    count+=1
    cnt[r]=count
    
    for i in graph[r]:
        if visited[i]==0:    
            dfs(v,e,i)

count=0
dfs(n,m,r)

for i in range(1,n+1):
    print(cnt[i])

File: 백준/Silver/24479. 알고리즘 수업 － 깊이 우선 탐색 1/알고리즘 수업 － 깊이 우선 탐색 1.swift

import Foundation




let input = readLine()!.split(separator: " ").map { Int($0)! }

let n = input[0] , m = input[1] , r = input[2]

var graph : Array<Array<Int>> = Array(repeating: [], count: n+1)
                                                                
var visited : Array<Bool> = Array(repeating: false, count: n+1)
    
var visitedWhen : Array<Int> = Array(repeating: 0, count: n+1)

var checker : Int = 0


for _ in 1...m {
    
    let input2 = readLine()!.split(separator: " ").map { Int($0)! }
    let a = input2[0] , b = input2[1]
    
    graph[a].append(b)
    graph[b].append(a)
    

}

for i in 1...n{
    graph[i].sort()
}



func dfs( now : Int ) {
    
    visited[now] = true
    checker += 1
    visitedWhen[now] = checker
    
    for i in graph[now] {
    
        if visited[ i ] == false {
        
            dfs(now: i)
        
        }
    }
}


dfs(now: r)

for i in 1...n{
    print(visitedWhen[i])
}
